zrcs/zrc-3.md

```diff
@@ -0,0 +1,431 @@
+| `CodeNonceError` | `Int32` | `-5` | Emit when the transition call tries to use a nonce lower than the accounts present nonce.
+| `CodeUnexpectedError` | `Int32` | `-6` | Emit when the transition call runs into an unexpected error.
+
+
+### C. Immutable Variables
+
+| Name                 | Type          | Description                                                           |
+| -------------------- | ------------- | --------------------------------------------------------------------- |
+| `contractOwner`      | `ByStr20`     | The owner of the contract initialized by the creator of the contract. |
+| `name`               | `String`      | The name of the fungible token.                                       |
+| `symbol`             | `String`      | The symbol of the fungible token.                                     |
+| `default_tellers`  | `List ByStr20`| The adddresses set as default for tellers all token holders.        |
+| `decimals`           | `Uint32`      | The number of decimal places a token can be divided by.               |
+
+### D. Mutable Fields
+
+| Name                | Type                                                              | Description                                                                                                                                                           |
+| ------------------- | ----------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `total_tokens` | `Uint128 = Uint128 0` |    Total amount of tokens.       |
+| `revokedDefaultTeller` | `Map ByStr20 (Map ByStr20 Bool) = Emp ByStr20 (Map ByStr20 Bool)` |    Mapping of `default_tellers` that have been revoked by token hodlers.       |
+| `balancesMap`     | `Map ByStr20 Uint128 = Emp ByStr20 Uint128`                       | Mapping between token owner to number of owned tokens.                                                                                                  |
+| `tellersMap`    | `Map ByStr20 (Map ByStr20 Bool) = Emp ByStr20 (Map ByStr20 Bool)`                       | Mapping between token owner to approved address. Token owner can approve an address (as an teller) to transfer tokens to other addresses.   |
+| `allowancesMap` | `Map ByStr20 (Map ByStr20 Uint128) = Emp ByStr20 (Map ByStr20 Uint128)` |    Mapping between token owner to approved address. Token owner can give an address an allowance of tokens to transfer tokens to other addresses.       |            
+| `accountNonce` | `Map ByStr20 Uint128 = Emp ByStr20 Uint128` |    Mapping between token owner to their present nonce. |
+
+### E. Transitions
+
+#### 1. ReauthorizeDefaultTeller
+
+```ocaml
+(* @dev: Re-authorize a default teller*)
+(* @param teller: Amount of tokens to be sent.       *)
```

Comment:
(* @param teller: Amount of tokens to be sent.       *) -> (* @param teller: The address of the `teller to be re-authorized. *)